kfs-core/src/main/java/edu/arizona/kfs/vnd/service/impl/EcustomsServiceImpl.java

```diff
@@ -0,0 +1,698 @@
+        FileOutputStream dataFileStream = new FileOutputStream(dataFile);
+        List<String> outputVendorLineList = new ArrayList<String>();
+//        List<VendorDetailExtension> updatedVDEList = new ArrayList<VendorDetailExtension>();
+//        List<Note> noteList = new ArrayList<Note>();
+
+        for (VendorDetail vendorDetail : vendorDetails) {
+            String vendorLine = generateVendorLine(vendorDetail);
+            if (!outputVendorLineList.contains(vendorLine)) {
+                outputVendorLineList.add(vendorLine);
+                dataFileStream.write(vendorLine.getBytes());
+//                VendorDetailExtension updatedVDE = updateVendorDetailExtension(vendorDetail);
+                updateVendor(vendorDetail);
+//                Note note = createNote(vendorDetail);
+//                updatedVDEList.add(updatedVDE);
+//                noteList.add(note);
+            }
+        }
+//        getBusinessObjectService().save(updatedVDEList);
+//        getBusinessObjectService().save(noteList);
+        dataFileStream.flush();
+        dataFileStream.close();
+        writeVendorCountFile(outputVendorLineList.size());
+        KualiInteger listSize = new KualiInteger(outputVendorLineList.size());
+        saveLastRunDate(jobRunDate, listSize);
+        return true;
+    }
+
+    /**
+     * write the file containing number of vendors processed for ecustoms
+     *
+     * @param vendorDetails
+     */
+    private void writeVendorCountFile(int size) throws IOException {
+        FileOutputStream countFileStream = new FileOutputStream(vendorCountFile);
+        String outputMessage = "Vendors added to ecustoms file = " + size;
+        LOG.info(outputMessage);
+        countFileStream.write(outputMessage.getBytes());
+        countFileStream.flush();
+        countFileStream.close();
+    }
+
+    /**
+     * Get the list of vendors that need to be processed since the last daily run.
+     *
+     * @param jobRunDate
+     * @return list of vendors to process for security compliance
+     * @throws WorkflowException
+     */
+    private List<VendorDetail> getDailyVendorDetailList(Date jobRunDate) {
+        List<VendorDetail> retval = new ArrayList<VendorDetail>();
+
+        List<VendorDetail> vendorDetails = new ArrayList<VendorDetail>();
+
+        List<VendorDetail> recentVendorList = getRecentVendorList();
+        List<VendorDetail> exportList = getVendorsWithExportFlagSetToNo();
+
+        vendorDetails.addAll(recentVendorList);
+        vendorDetails.addAll(exportList);
+
+        if (vendorDetails.isEmpty()) {
+            LOG.info("No vendor candidates found.");
+            return retval;
+        } else {
+            LOG.info("Found " + vendorDetails.size() + " total vendor candidates for ecustoms.");
+        }
+
+        for (VendorDetail vendorDetail : vendorDetails) {
+            boolean wasProcessedToday = hasProcessedToday(vendorDetail);

```

Comment:
This was done this way for debugging purposes. Even if the method call were to be put in the if statement, the memory space will still be allocated. Doing it in this way allows access to the value when debugging.